Include/internal/pycore_typeobject.h

```diff
@@ -62,6 +64,20 @@ _PyStaticType_GET_WEAKREFS_LISTPTR(static_builtin_state *state)
     return &state->tp_weaklist;
 }
 
+/* Like PyType_GetModuleState, but skips verification
+ * that type is a heap type with an associated module */
+static inline void *
+_PyType_GetModuleState(PyTypeObject *type)
```

Comment:
I trust your need for the API based on the discuss thread. The only question I have is if this should use the ALL_CAPS idiom to represent the "unchecked fast API" as other things such as `_PyObject_GET_WEAKREFS_LISTPTR` and `PyTuple_GET_ITEM` do.

`_PyType_GET_MODULE_STATE()` perhaps?

That makes the difference from PyType_GetModuleState more obvious than just the leading underscore.

Reply:
That's the naming convention for macros (see [PEP 7](https://peps.python.org/pep-0007/#naming-conventions)), and -- I suppose -- things that started out as macros but were converted to functions.
Using it to mean “unchecked fast API” surprised me a bit. Should we update PEP 7?

Reply:
Post PEP-670 it may be a good idea to update PEP-7, yeah. Pre PEP-670 this would probably have been a macro, I guess, so the naming kinda makes sense. I'm fine with both `_PyType_GetModuleState` and `_PyType_GET_MODULE_STATE`.

Reply:
FWIW, I don't really care either way -- it's private API. I was just surprised when it was brought up.

Reply:
If a function is exposed to the public C API, for "unsafe" flavor of a function, I would suggest adding "Unsafe" or "Unchecked" to its name rather than relying on lowercase/uppercase. Example of existing functions:

* \_PyThreadState_**Unchecked**Get()
* \_PyGILState_GetInterpreterState**Unsafe**()

For the specific case of the internal C API, so far, I just added ``_`` prefix: ``PyThreadState_GET()`` (slow, regular function call) becomes  ``_PyThreadState_GET()`` (fast, inlined). I'm not really excited by a single underscore, it's easy to miss. But well, that's the status quo convention :-)

Maybe we should just go for ``Fast`` suffix :-D ``_PyType_GetModuleStateFast()`` :-)